client.go

```diff
@@ -51,6 +53,24 @@ func WithTokenFile(path string) ClientOption {
 	return WithToken(t)
 }
 
+// WithBaseURL provides a method to set the base URL for standard API calls to differ
+// from the default.
+func WithBaseURL(url string) ClientOption {
+	return func(c *Client) error {
```

Comment:
Do we want to validate this url using `url.Parse`? Should we construct our URLs pragmatically [like this](https://github.com/uhthomas/tesla/blob/f00d0e07a2b61907751647920f5b7c1f8197c07a/vehicle.go#L235-L236)?

Reply:
Lot of effort for little benefit? If you set it wrong, you set it wrong ;)

Reply:
Hm, it's quite subjective. As a consumer, I would prefer my constructor to fail if the URL is invalid, rather than for every request to fail.

Reply:
I'm ambivalent about validating the URL here - I'd rather a consumer validate it themselves.

As for constructing the URLs, that's probably the right move but I'd rather that be done in an independent PR/change.